server/storage/gdrive.go

```diff
@@ -45,13 +46,20 @@ func NewGDriveStorage(clientJSONFilepath string, localConfigPath string, basedir
 	}
 
 	// If modifying these scopes, delete your previously saved client_secret.json.
-	config, err := google.ConfigFromJSON(b, drive.DriveScope, drive.DriveMetadataScope)
-	if err != nil {
-		return nil, err
+	var httpClient *http.Client
+
+	if strings.Contains(string(b), `"type": "service_account"`) {
```

Comment:
let's have a separate cli param for service account file

Reply:
param like --gdrive-type=service / oauth2?
or we can actually check in json.

Reply:
Oauth2
`
{
    "installed": {
        "client_id": "clientid",
        "project_id": "projectname",
        "auth_uri": "https://accounts.google.com/o/oauth2/auth",
        "token_uri": "https://oauth2.googleapis.com/token",
        "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
        "client_secret": "clientsecret"
    }
}
`

Service Account
`
{
  "type": "service_account",
  "project_id": "projectname",
  "private_key_id": "keyid",
  "private_key": "-----BEGIN PRIVATE KEY-----\nKEYSECRET\n-----END PRIVATE KEY-----\n",
  "client_email": "transfersh@projectname.iam.gserviceaccount.com",
  "client_id": "clientid",
  "auth_uri": "https://accounts.google.com/o/oauth2/auth",
  "token_uri": "https://oauth2.googleapis.com/token",
  "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
  "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/transfersh%40projectname.iam.gserviceaccount.com"
}
`

Reply:
> param like --gdrive-type=service / oauth2?
> or we can actually check in json.

param like `gdrive-client-json-filepath`: `gdrive-service-account-json-filepath`

only one of the two can be set

in the factory you can translate this to a single param with the path and a new param for the type (service account/oauth2): it is up to you